lib/elastic_record/errors.rb

```diff
@@ -4,11 +4,29 @@ class Error < StandardError
 
   class ConnectionError < Error
     attr_reader :status_code, :payload
-    def initialize(status_code, message, payload = nil)
+
+    def initialize(status_code, json_error, json_payload = nil)
       @status_code = status_code
-      message = "#{message} (for payload '#{payload}')" if payload
+
+      message = build_message(json_error, json_payload)
+
       super(message)
     end
+
+    private
+
+      def build_message(json_error, json_payload)
+        error   = { elasticsearch_response: parse_or_return(json_error) }
+        payload = { request_payload:        parse_or_return(json_payload) }
```

Comment:
If payload isn't provided, this will just return `nil`; so, no need to set a default.